src/main.cpp

```diff
@@ -212,6 +215,35 @@ static void handle_signal( int sig )
 	}
 }
 
+static void raise_fd_limit()
```

Comment:
- Instead of using an hardcoded number, it would be better to just bump the limit to the max.
- Please use early returns for error handling to avoid unnecessary indentation.
- We should restore the original limit when we fork/exec a child.

For an example of an implementation of `raise_fd_limit`, see https://github.com/swaywm/sway/pull/6629.